OWML.ModHelper.Menus/ModInputInput.cs

```diff
@@ -0,0 +1,138 @@
+﻿using OWML.Common.Menus;
+using OWML.ModHelper.Events;
+using OWML.Common;
+using UnityEngine;
+using UnityEngine.UI;
+using System;
+
+namespace OWML.ModHelper.Menus
+{
+    public class ModInputInput : ModInput<string>, IModInputInput
+    {
+
+        private string _value;
+        private Vector3 scale;
+        private UIStyleManager styleManager;
+        private HorizontalLayoutGroup _layoutGroup;
+        public IModLayoutButton Button { get; }
+        protected readonly IModInputMenu InputMenu;
+        protected readonly TwoButtonToggleElement ToggleElement;
+
+        public ModInputInput(TwoButtonToggleElement element, IModMenu menu, IModInputMenu inputMenu) : base(element, menu)
+        {
+            ToggleElement = element;
+            InputMenu = inputMenu;
+            OnChange += Upd;
+            scale = element.GetValue<Button>("_buttonTrue").transform.localScale;
+            Button = new ModLayoutButton(element.GetValue<Button>("_buttonTrue"), menu);
+            Button.OnClick += Open;
+            var noButton = ToggleElement.GetValue<Button>("_buttonFalse");
+            noButton.transform.parent.gameObject.SetActive(false);
+            _layoutGroup = Button.LayoutGroup;
+            ((RectTransform)_layoutGroup.transform).sizeDelta = new Vector2(((RectTransform)Button.Button.transform.parent).sizeDelta.x * 2, ((RectTransform)Button.Button.transform.parent).sizeDelta.y);
+            styleManager = MonoBehaviour.FindObjectOfType<UIStyleManager>();
+
+            var layoutGroup = Button.Button.transform.parent.parent.GetComponent<HorizontalLayoutGroup>();
+            layoutGroup.childControlWidth = true;
+            layoutGroup.childForceExpandWidth = true;
+            Button.Button.transform.parent.GetComponent<LayoutElement>().preferredWidth = 100;
+        }
+
+        private void Upd(string thing)
```

Comment:
Maybe more descriptive names? Even if it is private, it would still increase readability

Reply:
was mentioned by alek and already fixed